tests/test_crystal_graph.py

```diff
@@ -127,55 +126,57 @@ def test_crystal_graph_different_cutoff_fast():
 
 
 def test_crystal_graph_perturb_legacy():
-    np.random.seed(0)
     structure_perturbed = structure.copy()
-    structure_perturbed.perturb(distance=0.1)
+    fixed_rng = np.random.default_rng(0)
+    with patch("numpy.random.default_rng", return_value=fixed_rng):
+        structure_perturbed.perturb(distance=0.1)
```

Comment:
we should probably add a `seed` kwarg to `Structure.perturb` that's simply passed into `np.random.default_rng(seed=seed)` over in `pymatgen`

Reply:
That's a good point, especially with the generator implementation. 

I didn't realize this downside before but it turns out the new generator implementation every `rng` is a isolated instance, making it impossible to set a global seed and control all random states (or it's just me unaware of that).